flutter/lib/ui/history/result_filter_screen.dart

```diff
@@ -174,6 +191,23 @@ class _ResultFilterScreenState extends State<ResultFilterScreen> {
     );
   }
 
+  Widget _sortBy(ResultFilter filter) {
```

Comment:
The sort should not be dependent on the filter or part of the filter.

Reply:
Should we move then the sort by option out from the filter screen? And introduce the new sort screen with its state?

Reply:
Generally I think in our case the filtering and sorting are related and used together. We don't use the sorting separately in the other part of the app. Ofcourse we can separate it (it's easy to do, after the refactoring), but don't think it makes a lot of sense currently. If in the future we have the need to use it separately or have the more advanced sorting we can separate. Please let me know what do you think.

Reply:
> Specifically: Show the list of benchmark-specific results in the history screen (one level deeper than currently). Is this still valid?

Yes. It's still valid. The history screen should show a list of benchmark-specific results.

> now you are saying to not change the list's structure.

Not sure where I mentioned that. Maybe a misunderstanding?

Reply:
Ok, now is clear. So for all cases we need to show results by benchmarks, not only for the sorting by task throughput case. Correct? Was thinking we need to show 2 distinguished lists for each case separately. Thanks for the clarification. Will apply changes.

Reply:
Moved sort logic to a separate class;